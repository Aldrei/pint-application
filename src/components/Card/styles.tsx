import { styled } from '@mui/material/styles';


export const CardStyle = styled('div')(({ theme }) => ({
  padding: '40px',
  width: 'fit-content',
  borderRadius: '8px',
  backgroundColor: theme.palette.background.default,
  border: '1px solid rgb(23, 58, 94)',
  /* border: 1px solid blue; */
}));
