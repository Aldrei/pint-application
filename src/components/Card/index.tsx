import React from 'react';

import { CardStyle } from './styles';

interface Props {
  children?: React.ReactNode;
}

const Card = (props: Props) => {
  return <CardStyle className='cardComp'>{props.children}</CardStyle>;
};

export default Card;