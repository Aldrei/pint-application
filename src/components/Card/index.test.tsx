import { render } from '@testing-library/react';
import React from 'react';

import Card from './index';

describe('Card component', () => {
  it('Should render correctly', () => {
    const nodeEl = render(<Card><h1 className="child-test">Test</h1></Card>);

    const nodeElChild = nodeEl.baseElement.getElementsByClassName('child-test');
    expect(nodeElChild.length).toBe(1);
    
    expect(nodeEl.baseElement).toMatchSnapshot();
  });
});
