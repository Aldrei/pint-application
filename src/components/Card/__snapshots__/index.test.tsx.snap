// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component Should render correctly 1`] = `
<body>
  <div>
    <div
      class="cardComp css-p1o44d"
    >
      <h1
        class="child-test"
      >
        Test
      </h1>
    </div>
  </div>
</body>
`;
